lib/containers/utils.pm

```diff
@@ -78,6 +79,48 @@ sub registry_url {
     return sprintf("%s/%s:%s", $repo, $container_name, $version_tag);
 }
 
+# This is simple and universal
+sub runtime_smoke_tests {
+    my %args = @_;
+    my $runtime = $args{runtime};
+    my $image = $args{image} // registry_url('alpine', '3.6');
+
+    record_info('Smoke', "Smoke test running image: $image on runtime: $runtime.");
+
+    # Pull image from registry
+    if ($runtime =~ /nerdctl/) {
+        assert_script_run("$runtime image pull --insecure-registry $image");
+    } else {
+        assert_script_run("$runtime pull $image");
+    }
+
+    # List locally available images
+    # if we miss $image the test will fail later
+    assert_script_run("$runtime image ls");
+
+    # crictl is not implemented as it needs a lot of additional settings just to run container
+    if ($runtime !~ /crictl/) {
+        # Run container in foreground
+        assert_script_run("$runtime run -it --rm $image echo 'Hello'");
+
+        # Run container in background - it can take few seconds
+        assert_script_run("$runtime run -d --name 'sleeper' $image sleep 999");
```

Comment:
you can also do `"$runtime run -d --name 'sleeper' $image sh"` and it will hang forever

Reply:
I was not sure about this. We can improve it the next time.